src/wp-admin/includes/image-edit.php

```diff
@@ -1026,12 +1038,177 @@ function wp_save_image( $post_id ) {
+				$current_extension = pathinfo( $thumbnail_file, PATHINFO_EXTENSION );
+
+				// Create a file with then new extension out of the targeted file.
+				$target_file_name     = preg_replace( "/\.$current_extension$/", ".$extension", $thumbnail_file );
+				$target_file_location = path_join( $original_directory, $target_file_name );
+				$result               = $img->save( $target_file_location, $targeted_mime );
+
+				if ( is_wp_error( $result ) ) {
+					continue;
+				}
+
+				$subsized_images[ $targeted_mime ] = array( 'thumbnail' => $result );
+			} else {
+				$destination = trailingslashit( $original_directory ) . "{$filename}.{$extension}";
+				$result      = $img->save( $destination, $targeted_mime );
+
+				if ( is_wp_error( $result ) ) {
+					continue;
+				}
+
+				$main_images[ $targeted_mime ]     = $result;
+				$subsized_images[ $targeted_mime ] = $img->multi_resize( $resize_sizes );
+			}
+		}
+
+		foreach ( $mime_transforms as $targeted_mime ) {
+			// Make sure the path and file exists as those values are required.
+			$image_directory = null;
+			if ( isset( $main_images[ $targeted_mime ]['path'], $main_images[ $targeted_mime ]['file'] ) && file_exists( $main_images[ $targeted_mime ]['path'] ) ) {
+				// Add sources to original image metadata.
+				$meta['sources'][ $targeted_mime ] = array(
+						'file'     => $main_images[ $targeted_mime ]['file'],
+						'filesize' => wp_filesize( $main_images[ $targeted_mime ]['path'] ),
+				);
+				$image_directory                   = pathinfo( $main_images[ $targeted_mime ]['path'], PATHINFO_DIRNAME );
+			}
+
+			/**
+			 * If no original image was provided the image_directory can't be determined, in that scenario try to
+			 * find it from the `file` property.
+			 *
+			 * @see get_attached_file()
+			 */
+			if (
+					null === $image_directory
+					&& isset( $meta['file'] )
+					&& 0 !== strpos( $meta['file'], '/' )
+					&& ':\\' !== substr( $meta['file'], 1, 2 )
+			) {
+				if ( false === $uploads_dir['error'] && isset( $uploads_dir['basedir'] ) ) {
+					$file = path_join( $uploads_dir['basedir'], $meta['file'] );
+					if ( file_exists( $file ) ) {
+						$image_directory = pathinfo( $file, PATHINFO_DIRNAME );
+					}
+				}
+			}
+
+			if ( null === $image_directory ) {
+				continue;
+			}
+
+			foreach ( $meta['sizes'] as $size_name => $size_details ) {
+				if ( empty( $subsized_images[ $targeted_mime ][ $size_name ]['file'] ) ) {
+					continue;
+				}
+
+				// Add sources to resized image metadata.
+				$subsize_path = path_join( $image_directory, $subsized_images[ $targeted_mime ][ $size_name ]['file'] );
+				if ( ! file_exists( $subsize_path ) ) {
+					continue;
+				}
+
+				$meta['sizes'][ $size_name ]['sources'][ $targeted_mime ] = array(
+						'file'     => $subsized_images[ $targeted_mime ][ $size_name ]['file'],
+						'filesize' => wp_filesize( $subsize_path ),
+				);
```

Comment:
Same here, can we reduce the indentation?

```suggestion
				$meta['sizes'][ $size_name ]['sources'][ $targeted_mime ] = array(
					'file'     => $subsized_images[ $targeted_mime ][ $size_name ]['file'],
					'filesize' => wp_filesize( $subsize_path ),
				);
```